### ISSUE TITLE
Fix UVM assignment of event data types

### ISSUE THREAD

**Issue by wsnyder** [verilator/verilator#4468]
Fix UVM assignment of event data types.  This is one of a few remaining blockers to UVM #1538.

This can be seen by editing test_regress/t/t_uvm_all.pl to remove --debug-exit-uvm, and commenting out the line to workaround #4466.

```
%Error-UNSUPPORTED: t/t_uvm_pkg_all.vh:7475:25: Unsupported: assignment of event data type
%Error-UNSUPPORTED: t/t_uvm_pkg_all.vh:7496:25: Unsupported: assignment of event data type
 7496 |                 m_event = e.m_event;
      |                         ^
```

**Comment by wsnyder** [comment]
See #4403

**Comment by wsnyder** [comment]
Was fixed back with (#4403).
